scripts/generator/graphql_generator/csharp/SDK/iOS/iOSNativeCheckout.ApplePayEventReceiver.cs.erb

```diff
@@ -60,7 +60,76 @@ namespace <%= namespace %>.SDK.iOS {
+
+            Action<ShopifyError> respondToSetFinalCheckoutFieldsError = (ShopifyError error) => {
+                ApplePayEventResponse response = new ApplePayEventResponse(ApplePayAuthorizationStatus.Failure);
+
+                // Check to see if this is a recoverable user error
+                if (error.Type == ShopifyError.ErrorType.UserError) {
+                    var userErrors = CurrentCart.UserErrors;
+                    var status = GetAuthorizationStatusFromShippingUserErrors(userErrors);
+
+                    if (status == ApplePayAuthorizationStatus.InvalidShippingPostalAddress) {
+                        response = new ApplePayEventResponse(status, GetSummaryItems());
+                    } else if (status == ApplePayAuthorizationStatus.InvalidShippingContact) {
+                        response = new ApplePayEventResponse(status, GetSummaryItems(), GetShippingMethods());
+                    }
+                }
+
+                message.Respond(response.ToJsonString());
+            };
+
+            Action<ShopifyError> respondToCheckoutError = (ShopifyError error) => {
+                ApplePayEventResponse response = new ApplePayEventResponse(ApplePayAuthorizationStatus.Failure);
+
+                // Check to see if this is a recoverable user error
+                if (error.Type == ShopifyError.ErrorType.UserError) {
+                    var userErrors = CurrentCart.UserErrors;
+                    var status = GetAuthorizationStatusFromCheckoutUserErrors(userErrors);
+
+                    if (status != ApplePayAuthorizationStatus.Failure) {
+                        response = new ApplePayEventResponse(status, GetSummaryItems(), GetShippingMethods());
+                    } else {
+                        response = new ApplePayEventResponse(status);
+                    }
+                }
+
+                message.Respond(response.ToJsonString());
+            };
+
+            Action performCheckout = () => {
+                CurrentCart.CheckoutWithTokenizedPayment(tokenizedPaymentInput, (ShopifyError error) => {
+                    if (error == null) {
+                        message.Respond(new ApplePayEventResponse(ApplePayAuthorizationStatus.Success).ToJsonString());
+                    } else {
+                        respondToCheckoutError(error);
+                    }
+                });
+            };
+
+            ShippingFields? shippingFields = null;
+            if (checkout.requiresShipping()) {
+                shippingFields = new ShippingFields(payment.ShippingAddress, payment.ShippingIdentifier);
+            }
+
+            CurrentCart.SetFinalCheckoutFields(payment.Email, shippingFields, (ShopifyError error) => {
+                if (error == null) {
+                    performCheckout();
+                } else {
+                    respondToSetFinalCheckoutFieldsError(error);
+                }
+            });
```

Comment:
The length of this method and the number of callbacks we're inlining makes me a bit uneasy but I'm not sure splitting it up will make sense since it's mostly code related to flow. Does it make sense to split this up into two helper methods for setting the final checkout fields and doing the checkout?